### PR TITLE
Scale9Sprite cropped sprite frames support

### PR DESCRIPTION
**Pull request by newnon**
Scale9Sprite now fully support cropped sprite frames in both mode simple & slice

### CODE REVIEW

**Comment by newnon** [comment]
closed https://github.com/cocos2d/cocos2d-x/issues/14549

**Comment by zilongshanren** [comment]
@newnon

:+1:   Thanks for the refinement. 
Does scale9sprite support spriteframe trim now?

BTW, the PR broke the CI:

```
[aapt] /android_slave/workspace/ccs-pr/node/android/tests/cpp-tests/proj.android/assets/Images/blocks9с.png: error: Invalid filename. Unable to add.
```

File: cocos/ui/UIScale9Sprite.cpp [hunk 1]
```diff
@@ -503,23 +504,31 @@ namespace ui {
 
             auto capInsets = CC_RECT_POINTS_TO_PIXELS(_capInsetsInternal);
             auto textureRect = CC_RECT_POINTS_TO_PIXELS(_spriteRect);
-            auto spriteRectSize = CC_SIZE_POINTS_TO_PIXELS(_originalSize);
+            auto spriteRectSize = _spriteRect.size;
+            auto originalSize = CC_SIZE_POINTS_TO_PIXELS(_originalSize);
+            auto offset = CC_POINT_POINTS_TO_PIXELS(_offset);
+            
+            Vec4 offsets;
+            offsets.x = offset.x + (originalSize.width - textureRect.size.width) / 2;
```
Comment (zilongshanren) [inline cocos/ui/UIScale9Sprite.cpp:512]: I don't quite understand the `offset.z` and `offset.w`  mean, could you explain it? Thanks.

**Comment by zilongshanren** [comment]
image

This test effect is wrong.

**Comment by zilongshanren** [comment]
@newnon 
It breaks some of the tests in UI and Scale9sprite test, please check.

File: cocos/ui/UIScale9Sprite.cpp (hunk 1 above)
Comment (newnon) [inline cocos/ui/UIScale9Sprite.cpp:512]: offsets contain sizes of spriteFrame cropped part
x - left
y - top
z - right
w - bottom

**Comment by newnon** [comment]
I changed test a little, the upper scale9sprite loads from spriteframe bottom from file they must be the same, if you found some test broken please write which one, because i don't see any broken tests.

**Comment by zilongshanren** [comment]
@newnon
Does your PR support trim mode of Texture Packer?

**Comment by newnon** [comment]
Exactly, it is first goal of my PR

File: cocos/ui/UIScale9Sprite.cpp (hunk 1 above)
Comment (zilongshanren) [inline cocos/ui/UIScale9Sprite.cpp:512]: @newnon  I think it's better to add comments here or change the variable naming.
It's not intuitive at the first glance.

**Comment by zilongshanren** [comment]
image

**Comment by zilongshanren** [comment]
image

**Comment by zilongshanren** [comment]
@newnon 

Thanks. please take a look at the above two pictures, the rendering effect is wrong.

**Comment by newnon** [comment]
Ok i will look at this two tests

**Comment by newnon** [comment]
Fixed

**Comment by zilongshanren** [comment]
@newnon 
Thanks :+1:

**Pull request merged**